src/Export/DynamicProductGroupService.php

```diff
@@ -218,10 +246,109 @@ private function buildCriteria(): Criteria
         return $criteria;
     }
 
-    private function getCacheItem(): CacheItemInterface
+    private function getExportPaginationCacheItem(): CacheItemInterface
+    {
+        $id = sprintf('%s_%s_%s', self::CACHE_ID_PRODUCT_GROUP, $this->shopkey, $this->start);
+
+        return $this->cache->getItem($id);
+    }
+
+    private function setTotalInCache(CacheItemInterface $cacheItem, $total): void
+    {
+        $cacheItem->set($total);
+        $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+        $this->cache->save($cacheItem);
+    }
+
+    private function getDynamicGroupsTotalCacheItem(): CacheItemInterface
     {
-        $id = sprintf('%s_%s', self::CACHE_ID_PRODUCT_GROUP, $this->shopkey);
+        $id = sprintf('%s_%s_total', self::CACHE_ID_PRODUCT_GROUP, $this->shopkey);
 
         return $this->cache->getItem($id);
     }
+
+    /**
+     * Gets the total count of dynamic product groups from cache
+     */
+    public function getDynamicProductGroupTotalFromCache(): int
+    {
+        $cacheItem = $this->getDynamicGroupsTotalCacheItem();
+        if ($cacheItem && $cacheItem->isHit()) {
+            $cacheItem->expiresAfter(self::CACHE_LIFETIME_PRODUCT_GROUP);
+            $this->cache->save($cacheItem);
+
+            return (int)$cacheItem->get();
```

Comment:
When is this not an int? Can the cache not handle numbers?

Reply:
Yes it can, but the return value is a `mixed` value, so explicitly typecasting it into an int

Reply:
How can this be `mixed`? Will it return a string or why are we casting this to an int?